File: app/src/main/java/xyz/farshad/vocab/activity/WordListActivity.java
package xyz.farshad.vocab.activity;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import java.util.List;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import xyz.farshad.vocab.R;
import xyz.farshad.vocab.component.DataAdapter.WordListAdapter;
import xyz.farshad.vocab.model.Word;

public class WordListActivity extends AppCompatActivity {

    List<Word> words;
    private Long levelId;
    private String levelName;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_word_list);
        setTitle("Words");

        Bundle b = getIntent().getExtras();
        if (b != null && b.containsKey("levelId")) {
            levelName = b.getString("levelName");
            levelId = b.getLong("levelId");
            words = Word.find(Word.class, "level_id = ?", levelId.toString());

            showWordList(false);
        }
        setToolBar();
    }

    private void setToolBar(){
        ActionBar toolbar = getSupportActionBar();
        toolbar.setTitle("Words of " + levelName);
    }

    private void showWordList(boolean reload){
        if (reload){
            words = Word.find(Word.class, "level_id = ?", levelId.toString());
        }
        ArrayAdapter<Word> adapter = new WordListAdapter(WordListActivity.this, R.layout.word_list_view, words);
        ListView list = findViewById(R.id.wordMainListView);
        list.setAdapter(adapter);
        list.setItemsCanFocus(true);
    }
}

File: app/src/main/java/xyz/farshad/vocab/activity/WordListActivity.kt
package xyz.farshad.vocab.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import org.koin.androidx.viewmodel.ext.android.viewModel
import xyz.farshad.vocab.R
import xyz.farshad.vocab.viewmodel.WordViewModel

class WordListActivity : AppCompatActivity() {
    private val wordViewModel: WordViewModel by viewModel()
    private var levelId: Int? = null
    var isReview: Boolean = false
    private var levelName: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_word_list)
        title = "Words"

        val b = intent.extras
        if (b != null) {
            levelName = b.getString("levelName")
            isReview = b.getBoolean("isReview")

            if (isReview) {
                wordViewModel.fetchReviewWords()
            } else {
                levelId = b.get("levelId").toString().toInt()
                wordViewModel.findByLevelId(levelId!!)
            }
        }
        setToolBar()
        setObserver()
    }

    private fun setObserver() {
    }

    private fun setToolBar() {
        val toolbar = supportActionBar
        toolbar!!.title = "Words of " + levelName!!
    }
}
